cli/flox/src/commands/environment.rs

```diff
@@ -529,6 +550,143 @@ impl List {
                     pkgs.iter().for_each(|pkg| println!("{}", pkg));
                 }
             },
+            ListMode::Extended => self.print_extended(&flox, &mut *env)?,
```

Comment:
suggestion nonblocking: build for `NameOnly` instead of just using the manifest, drop `list_packages(&str)`, and always use the lockfile as the source of truth. And then if you do that you can factor out all the building code. I think it's surprising that the source of truth is different based on whether you pass `--name` or not.

Reply:
yeah youre right, we should do the build (lock optimally but thats not exposed currently) for all, will change